### ISSUE TITLE
very slow training

### ISSUE THREAD

**Issue by businiaoo** [yzd-v/FGD#43]
Line 86 in the forward function in mmdet/distillation/losses/fgd.py There are two for loops here, in my test, these two for loops slow down the training seriously, is there any solution?

**Comment by HUALIMUGU** [comment]
The for loops is build upon batch size, I guess there should not be some great slowing, I mean there are just 16 samples in total.

**Comment by Muke6** [comment]
I also encountered the same problem, have you solved it?

**Comment by Tongfengyu** [comment]
Is there need to add a line ' with paddle.no_grad(): ' when get the mask?

**Comment by Tongfengyu** [comment]
Is there need to add a line ' with torch.no_grad(): ' when get the mask?
